src/main/java/com/microsoft/sqlserver/jdbc/ConfigRetryRule.java

```diff
@@ -0,0 +1,279 @@
+                    }
+                } else if (timings[1].contains(PLUS_SIGN)) {
+                    String[] initialAndChange = timings[1].split("\\+");
+                    checkParameterIsNumeric(initialAndChange[0]);
+
+                    initialRetryTime = Integer.parseInt(initialAndChange[0]);
+                    operand = PLUS_SIGN;
+                    if (initialAndChange.length > 1) {
+                        checkParameterIsNumeric(initialAndChange[1]);
+                        retryChange = Integer.parseInt(initialAndChange[1]);
+                    } else {
+                        retryChange = 2;
+                    }
+                } else {
+                    checkParameterIsNumeric(timings[1]);
+                    initialRetryTime = Integer.parseInt(timings[1]);
+                }
+            } else if (timings.length > 2) {
+                MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_InvalidParameterFormat"));
+                Object[] msgArgs = {rule[1], "Too many arguments"};
+                throw new SQLServerException(null, form.format(msgArgs), null, 0, true);
+            }
+
+            if (rule.length == 3) {
+                retryQueries = (rule[2].equals(ZERO) ? "" : rule[2].toLowerCase());
+            }
+        } else {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_InvalidRuleFormat"));
+            Object[] msgArgs = {rule.length};
+            throw new SQLServerException(null, form.format(msgArgs), null, 0, true);
+        }
+    }
+
+    private void calculateWaitTimes() {
+        for (int i = 0; i < retryCount; ++i) {
+            int waitTime = initialRetryTime;
+            if (operand.equals(PLUS_SIGN)) {
+                for (int j = 0; j < i; ++j) {
+                    waitTime += retryChange;
+                }
+            } else if (operand.equals(MULTIPLICATION_SIGN)) {
+                for (int k = 0; k < i; ++k) {
+                    waitTime *= retryChange;
+                }
+            }
+            waitTimes.add(waitTime);
+        }
+    }
+
+    /**
+     * Returns the retry error for this ConfigRetryRule object.
+     *
+     * @return
+     *         The retry error
+     */
+    public String getError() {
+        return retryError;
+    }
+
+    /**
+     * Returns the retry error for this ConfigRetryRule object.
+     *
+     * @return
+     *         The retry error
+     */
+    public String getOperand() {
+        return operand;
+    }
+
+    /**
+     * Returns the retry error (errors to retry on) for this ConfigRetryRule object.
+     *
+     * @return
+     *         The retry error
+     */
+    public int getInitialRetryTime() {
+        return initialRetryTime;
+    }
+
+    /**
+     * Returns the retry change (timing change to apply to wait times) for this ConfigRetryRule object.
+     *
+     * @return
+     *         The retry change
+     */
+    public int getRetryChange() {

```

Comment:
this is not used anywhere outside of this class, why does it need to be public? as a matter of fact, why does this need to exist at all if it's just within this class